radix-engine-interface/src/model/resource/non_fungible_local_id.rs

```diff
@@ -70,55 +70,71 @@ impl From<u128> for NonFungibleLocalId {
 /// Represents the local id of a non-fungible.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum NonFungibleLocalId {
+    /// String of `[_0-9a-zA-Z]{1,64}`.
     String(String),
-    Number(u64),
+    /// Unsigned integers, up to u64.
+    Integer(u64),
+    /// Bytes, of length between 1 and 64.
     Bytes(Vec<u8>),
+    /// UUID, v4, variant 1, big endian. See https://www.rfc-editor.org/rfc/rfc4122
     UUID(u128),
 }
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum IdValidationError {
+pub enum ContentValidationError {
     TooLong,
     Empty,
     ContainsBadCharacter(char),
+    NotUuidV4Variant1,
 }
 
 impl NonFungibleLocalId {
     pub fn id_type(&self) -> NonFungibleIdType {
         match self {
             NonFungibleLocalId::String(..) => NonFungibleIdType::String,
-            NonFungibleLocalId::Number(..) => NonFungibleIdType::Number,
+            NonFungibleLocalId::Integer(..) => NonFungibleIdType::Integer,
             NonFungibleLocalId::Bytes(..) => NonFungibleIdType::Bytes,
             NonFungibleLocalId::UUID(..) => NonFungibleIdType::UUID,
         }
     }
 
-    pub fn validate_contents(&self) -> Result<(), IdValidationError> {
+    pub fn validate_contents(&self) -> Result<(), ContentValidationError> {
         match self {
             NonFungibleLocalId::String(value) => {
                 if value.len() == 0 {
-                    return Err(IdValidationError::Empty);
+                    return Err(ContentValidationError::Empty);
                 }
                 if value.len() > NON_FUNGIBLE_LOCAL_ID_MAX_LENGTH {
-                    return Err(IdValidationError::TooLong);
+                    return Err(ContentValidationError::TooLong);
                 }
                 for char in value.chars() {
                     if !matches!(char, '0'..='9' | 'A'..='Z' | 'a'..='z' | '_') {
-                        return Err(IdValidationError::ContainsBadCharacter(char));
+                        return Err(ContentValidationError::ContainsBadCharacter(char));
                     }
                 }
                 Ok(())
             }
             NonFungibleLocalId::Bytes(value) => {
                 if value.len() == 0 {
-                    return Err(IdValidationError::Empty);
+                    return Err(ContentValidationError::Empty);
                 }
                 if value.len() > NON_FUNGIBLE_LOCAL_ID_MAX_LENGTH {
-                    return Err(IdValidationError::TooLong);
+                    return Err(ContentValidationError::TooLong);
                 }
                 Ok(())
             }
-            NonFungibleLocalId::Number(_) | NonFungibleLocalId::UUID(_) => Ok(()),
+            NonFungibleLocalId::Integer(_) => Ok(()),
+            NonFungibleLocalId::UUID(v) => {
+                // 0100 - v4
+                // 10 - variant 1
+                if (v & 0x00000000_0000_f000_C000_000000000000u128)
```

Comment:
Super picky, feel free to ignore - but should we have consistent casing? so `f` and `c` perhaps?

Reply:
Shame!